### ISSUE TITLE
[20210912] Weekly AI ArXiv 만담

### ISSUE THREAD

**Issue by jungwoo-ha** [jungwoo-ha/WeeklyArxivTalk#24]
- News
  - AAAI, CHI submission 완료 수고하셨습니다.
  - [주요기업, 연구소, 정부 합해 초대규모AI 공동 협력](https://news.naver.com/main/read.naver?mode=LSD&mid=shm&sid1=105&oid=015&aid=0004601530)
  - [AI 중심의 글로벌 합종연횡 - 우리의 대응은?](http://www.aitimes.com/news/articleView.html?idxno=140585)

 
- ArXiv
  - [CTAL: Pre-training Cross-modal Transformer for Audio-and-Language Representations](https://arxiv.org/abs/2109.00181)
    - Text와 Audio를 함께 pretraining 하는 multimodal transformer (CTAL)
    - [ST-BERT](https://ieeexplore.ieee.org/abstract/document/9414558) (ICASSP2021) 유사한 구성이지만 MLM + Masked Cross modal acousting modeling 으로 mel 복원 및 더욱 다양한 downstream task
    - Emotion classification, Sentimenty analysis, Speaker verification 등
![image](https://user-images.githubusercontent.com/11782739/132927531-5f077803-77aa-46a4-b2e9-19a0729e2008.png) 
![image](https://user-images.githubusercontent.com/11782739/132927717-9747fa05-b99c-4eb7-a6f7-2cc23fb1ce6e.png)

  - [Datasets: A Community Library for Natural Language Processing](https://arxiv.org/abs/2109.02846v1)
    - Huggingface에서 주요 NLP 데이테셋 집대성한 library
    - 650 개 이상의 데이터셋, 250명 이상의 contributor
    - 데이터셋과 관련된 여러가지 통계치도 공개함. 랭킹에서 NLP연구 순위대비 공개데이터 순위가 낮음 (NLP학회 논문순위는 10위권인데 영어빼고 18위, 중국은 19위)
    - https://github.com/huggingface/datasets
![image](https://user-images.githubusercontent.com/11782739/132927990-9de9a77a-2533-4a3d-a325-60978b9dc546.png)

  - [Finetuned Language Models Are Zero-Shot Learners](https://arxiv.org/abs/2109.01652v1)
    - 137B PLM + Instruction tune --> FLAN (from Google Brain)
    - 이렇게 학습시킨 모델이 다른 unseen task에서의 zeroshot 에서 뛰어난 성능
    - 175B GPT3 zero-shot / few-shot 과 비교.
    - 추가 학습 오버헤드를 고려하면 p-tuning과 같은 prompt-optimized GPT3와 비교를 해야할 거 같은데... (구글 정도면 repro해서 해볼 수 있을 것 같은데...)
![image](https://user-images.githubusercontent.com/11782739/132929124-30a65037-6064-478b-bb85-01c36b1f996d.png)
![image](https://user-images.githubusercontent.com/11782739/132929141-d6b96f6d-29e5-461b-90f3-dfca38160b39.png)
![image](https://user-images.githubusercontent.com/11782739/132929174-1147f7df-0243-41ce-bf2a-4bcee6f24198.png)

    
  - [Transformers in Vision: A Survey](https://arxiv.org/abs/2101.01169v3)
    - 제목이 내용을 그대로 보여줌.
    - 4월에 초안이 나온 이후 계속 수정 중 (아마도 리뷰중?). 그래서 그런지 3월말 이후 나온 최신 Vision Transformer류의 연구는 누락
    - image recogntion backbone 류 뿐 아니라 생성모델, low-level vision, 기존 모델에 SA를 포함하는 연구까지 다양하게 망라
    - 그럼에도 불구하고 수업&정리 용으로는 꽤 좋을 것 같음
![image](https://user-images.githubusercontent.com/11782739/132930389-068423ce-ad25-43e9-9951-06ee67ed4074.png)
 
  - [Textless NLP: Generating expressive speech from raw audio](https://ai.facebook.com/blog/textless-nlp-generating-expressive-speech-from-raw-audio)
    - Text 없이 음성-to-음성으로 언어를 학습하는 모델 GSLM (from Facebook AI)
    - 문자 != 언어 임을 확인시켜주는 연구 
    - 음성 인식과 합성과 관련된 다양한 downstream task에서 뛰어난 성능
    - Vector quantized feature의 효용을 다시한번 확인
    - 음색/운율 변환 음성합성 적용: https://speechbot.github.io/pgslm/
![image](https://user-images.githubusercontent.com/11782739/132930580-c9bc2c68-8a4b-4d24-b06b-e9be4f6e3d23.png)

  - [Pose with Style: Detail-Preserving Pose-Guided Image Synthesis with Conditional StyleGAN](https://pose-with-style.github.io/)
    - StyleGAN2 기반의 디테일이 매우 잘 보존되는 포즈 변환 생성 모델 (from Adobe Research, SIGGRAPH Asia 2021)
    - 옷만 변경도 가능 
    - DensePose 써서 포즈정보 추출, coordiate 위해 mesh grid정보도 활용해서 StyleGAN generator 에 컨디셔닝
![image](https://user-images.githubusercontent.com/11782739/132930794-0732f52a-97c7-4f4a-aff4-7a3e9b709cd7.png)
![image](https://user-images.githubusercontent.com/11782739/132930853-13badf44-c16d-4718-b125-fd16ff3ab206.png)
 
  - [CodeT5: Identifier-aware Unified Pre-trained Encoder-Decoder Models for Code Understanding and Generation](https://arxiv.org/abs/2109.00859v1)
    - T5의 Code 버전 (from Saleforce Research Asia + NTU, EMLNP 2021)
    - code summarization, NL2code, code 언어변환 등
    - CODEXGLUE 에서 평가 (CODEX는 concurrent 이므로 패스)
    - https://blog.einstein.ai/codet5/
    - https://github.com/salesforce/CodeT5
![image](https://user-images.githubusercontent.com/11782739/132977867-2611aa4f-776b-4bc1-9766-25dc90a54055.png)
![image](https://user-images.githubusercontent.com/11782739/132977906-cbea2d4b-4f82-4e25-8441-fb7d0a7f0560.png)
![image](https://user-images.githubusercontent.com/11782739/132977923-503fd141-68ce-48a7-b59a-7229a5e4ac96.png)

**Comment by hollobit** [comment]
NIA, 국내 공공기관 최초로 「국가 인공지능 사업추진 윤리원칙」제정
>- 인공지능 사업을 추진하는 과정에서 지켜야 할 3단계 윤리원칙 선포 -
>* https://zdnet.co.kr/view/?no=20210907102042
>* https://www.nia.or.kr/site/nia_kor/ex/bbs/View.do?cbIdx=90549&bcIdx=23786&parentSeq=23786
![NIA ethics](https://image.zdnet.co.kr/2021/09/07/590c344e75bce794437435dc1e74f6d8.jpg)

인공지능(AI)은 진정 발명자가 될 수 없나...'다부스 특허'를 둘러싼 논란의 속내는
>* http://www.aitimes.com/news/articleView.html?idxno=140557

ML Reproducibility Challenge 2021
> 다섯 번째 ML 재현성 챌린지. 이 행사의 주요 목표는 신뢰할 수 있고 재현 가능한 과학적 결과의 출판 및 공유를 장려하는 것
> 9개 컨퍼런스로 대상 확대 : NeurIPS , ICML , ICLR , ACL-IJCNLP , EMNLP, CVPR , ICCV , AAAI 및 IJCAI 
> 챌린지 발표 : 2021년 8월 31일
> 챌린지가 실시간으로 진행됩니다: 2021년 8월 31일
> 제출 마감(동료 검토 고려 예정) : 2022년 2월 4일(오후 11시 59분 AOE)
> 저널 특별호 저자 공고 마감일: 2022년 5월 15일
>* https://paperswithcode.com/rc2021

NEURIPS DATA-CENTRIC AI WORKSHOP (12/14)
>* http://datacentricai.org/ 
>* https://analyticsindiamag.com/andrew-ng-announces-the-launch-of-neurips-data-centric-ai-workshop/

Neural Algorithmic Reasoning
>* https://arxiv.org/abs/2105.02761
>* https://venturebeat.com/2021/09/10/deepmind-aims-to-marry-deep-learning-and-classic-algorithms/
![화면 캡처 2021-09-12 200416](https://user-images.githubusercontent.com/998803/132986524-7b68ad1f-b232-4de0-9aef-64c671803009.png)

Phe2vec: Automated disease phenotyping based on unsupervised embeddings from electronic health records
>* https://www.cell.com/patterns/fulltext/S2666-3899(21)00185-9
>* https://www.marktechpost.com/2021/09/11/phe2vec-an-automated-ai-framework-based-on-neural-networks-for-ehr-based-disease-phenotyping/
![ph2vec](https://els-jbs-prod-cdn.jbs.elsevierhealth.com/cms/attachment/e8e83cdc-11d8-405f-b624-2c8a7b4e95fd/gr2.jpg)

**Comment by laray5959** [comment]
## Paper

- [Text-Free Prosody-Aware Generative Spoken Language Modeling)](https://arxiv.org/abs/2109.03264)

    - Sample page : https://speechbot.github.io/pgslm/
    - Task : Prosody Transfer 
    - Motivation
        - 음성 분야에서는 ASR이나 speaker verification과 같은 task에서  pre-training을 사용하는 경우가 많음
        - 합성 분야에서는 generative spoken language modelling(GSLM)가 유일하게 GPT-2와 같이 접근방식
        - GSLM 는 text 대신 phone-like unit을 사용, prosody 정보까지 잃음
        - 이 논문에서는 expressive 한 운율을 표현하는 음성을 생성하기 위하여 prosody를 phonetic content와 모델링 함
        - a prosody-aware generative spoken language model (pGSLM)

    - Method
        - GPT의 speech version으로 unlabeled speech data를 사용하고 text를 사용하지 않음 
        - Phonetic and Prosodic Representations 
            - phonetic representation을 위하여 HuBERT 로부터 vocab size 100개의 unit을 사용 
            - speech waveform은 초당 50개의 unit sequence로 변환하고 unit duration과 F0를 사용 
            - F0는 prosodic information을 갖고 있지만 speaker information도 갖고 있음
                -  speaker-mean normalized log F0를 사용 


![image](https://github.com/laray5959/FigureList/blob/main/Text-Free%20Prosody-Aware%20Generative%20Spoken%20Language%20Modeling.PNG?raw=true)

**Comment by veritas9872** [comment]
**PermuteFormer: Efficient Relative Position Encoding for Long Sequences**

GitHub: https://github.com/cpcp1998/PermuteFormer

Arxiv: https://arxiv.org/pdf/2109.02377.pdf

![Screenshot (61)](https://user-images.githubusercontent.com/33523965/132987972-8445d014-e5fe-4e17-8aae-6da2dc7947ec.png)

Efficient Transformer의 구조 중 하나인 Performer에서 relative positional encoding을 사용하는 방법에 대해 북경대학교에서 논문이 나왔습니다.
Performer는 kernel을 사용해서 self-attention을 직접 계산하지 않음으로써 매우 긴 sequence를 연산할 때 linear scaling을 할 수 있었으나 positional encoding이 여전히 많은 parameter를 차지한다는 문제점이 있었습니다.
본 연구에서는 key와 query에 대해 permutation을 가함으로써 relative positional encoding을 performer에서도 사용할 수 있도록 했다는데에 의의가 있습니다.


**Self-supervised Tumor Segmentation through Layer Decomposition**

Website: https://xiaoman-zhang.github.io/Layer-Decomposition/

![image](https://user-images.githubusercontent.com/33523965/132988257-329fa21a-cb96-4b59-882e-e9b0aedb3c39.png)


Self-supervised learning으로 (거의) zero-shot learning을 할 수 있는 방법이 제시되었습니다.
본 논문에서는 기존 medical image에서 copy-paste와 유사하게 tumor 영상을 mask에 따라 synthetic image를 만든 후 그 image를 decomposition을 하도록 학습을 유도해 기존에 비해 효율적인 segmentation을 학습하도록 합니다.

가장 흥미로운 점은 zero-shot segmemtation에 적용을 했을 때에도 다른 방법은 random output을 생성하지만 layer-decomposition 방법은 나쁘지 않은 성능의 결과를 보여준다는 점입니다.


**Scaled ReLU Matters for Training Vision Transformers**

Arxiv: https://arxiv.org/pdf/2109.03810.pdf

![Screenshot (62)](https://user-images.githubusercontent.com/33523965/132988350-2499cccc-9c62-4b8a-b3b3-5be70e5123c2.png)

알리바바에서 나온 최신 논문으로 기존 ViT의 학습이 원활하지 않아 SAM, input layer freezing 등 방법들에 이어 scaled ReLU를 제시합니다.

매우 단순한 방법이지만 저자들에 의하면 ReLU 다음에 learned scaling factor 하나만으로도 기존 ViT에 비해 학습을 원활히 할 수 있다는 것을 실험적으로 보여줍니다.

**Comment by Clyde21c** [comment]
* [The Sensory Neuron as a Transformer: Permutation-Invariant Neural Networks for Reinforcement Learning](https://arxiv.org/abs/2109.02869)
  * Demo page: https://attentionneuron.github.io/
  * World model로 알려진 Google Brain의 David Ha 연구
  * Cellular system의 각 센서 입력에 대해 permutation invariant한 RL agent를 기존 policy에 AttentionNeuron 레이어을 추가하여 학습
  *  센서 시스템이 다이나믹하게 변하는 대규모 application 등 에 활용 가능
![스크린샷, 2021-09-12 22-00-46](https://user-images.githubusercontent.com/35162035/132988545-e24c63e5-9f4f-4394-8ed9-1368024806d5.png)
![스크린샷, 2021-09-12 22-00-36](https://user-images.githubusercontent.com/35162035/132988548-be9bdae8-cd0c-4b96-8595-54fcdef1697a.png)

**Comment by Kyung-Min** [comment]
Paper
- [Multi-Task Self-Training for Learning General Representations](https://arxiv.org/pdf/2108.11353.pdf)
  - ![image](https://user-images.githubusercontent.com/9531308/132989225-fc28badf-a8f4-4515-8827-5f1ab878dae3.png)
  - multi-task learning을 위해 여러개의 teacher networks가 unlabeld images에 다양한 pseudo label을 생성 (object detection, semantic segmentation 등)
  - Student networks가 pseudo label을 multi-task learning으로 학습
  - 기존 contrastive self-supervised learning보다 더 좋은 성능을 얻음 (이미지넷에 평가, 학습에는 같은 데이터 사용, simclr)
  - scale이 커질 때 contrastive self-supervised learning의 효과가 떨어지지만 self-training은 그렇지 않음 (JFT-300M)
  - pseudo label을 추가로 학습에 사용하면, 단순 multi-task supervised learning보다 더 좋은 성능을 얻을 수 있었음
